lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -302,6 +302,29 @@ protected function getModifiers()
     {
         return 'i';
     }
+    
+    /**
+     * Checks if given value was catched using getCatchablePatterns key
+     *
+     * @example getCatchablePatterns: ['[a-z]']
+     *          => isCaughtByKey('a', 0): true
+     *
+     * @example getCatchablePatterns: ['some_key' => '[a-z]']
+     *          => isCaughtByKey('c', 'some_key'): true
+     * @return bool
+     */
+    public function isCaughtByKey($value, $key)
+    {
+        if (!isset($this->getCatchablePatterns()[$key]) || !$value || !$this->input) {
+            return false;
+        }
+
+        $flags   = PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_OFFSET_CAPTURE;
+        $regex = sprintf('/(%s)/%s', $this->getCatchablePatterns()[$key], $this->getModifiers());
+        $matches = preg_split($regex, $this->input, -1, $flags);
```

Comment:
`preg_split` can return `false` in case of failure, in which case line 326 will crash. Please either assert that `$matches` is not `false`, or handle `false` (throw an exception with `preg_last_error()`). That should fix the phpstan part of the build.

Reply:
This case should also be caught by a test.